packages/contract/index.js

```diff
@@ -2,6 +2,17 @@ const Schema = require("@truffle/contract-schema");
 const Contract = require("./lib/contract");
 const truffleContractVersion = require("./package.json").version;
 
+// we need to make sure this function exists so ensjs doesn't complain as it requires
+// getRandomValues for some functionalities - webpack strips out the crypto lib
+// so we shim it here
+if (global.crypto && !global.crypto.getRandomValues) {
```

Comment:
This shimming of the `getRandomValues` function is duplicated a couple of times in the codebase. It is caused by @ensdomains/ensjs and hopefully we can get off that lib soon and get rid of these. Maybe we should leave a note in the comments accompanying all these shims that we can get rid of it after that dep is gone.